src/test/ensemble_flow_custom_node_tests.cpp

```diff
@@ -3302,6 +3409,124 @@ TEST_F(EnsembleFlowCustomNodeAndDynamicDemultiplexerLoadConfigThenExecuteTest, J
     this->checkResponse("pipeline_output", response, expectedOutput, {dynamicDemultiplyCount, 1, 10});
     tensor_map_t inputs, outputs;
 
+    auto pipelineDefinition = manager.getPipelineFactory().findDefinitionByName(pipelineName);
+    ASSERT_EQ(pipelineDefinition->getInputsInfo(inputs, manager), StatusCode::OK);
+    ASSERT_EQ(pipelineDefinition->getOutputsInfo(outputs, manager), StatusCode::OK);
+    ASSERT_NE(inputs.find(pipelineInputName), inputs.end());
+    ASSERT_NE(outputs.find(pipelineOutputName), outputs.end());
+    auto& input_A = inputs.at(pipelineInputName);
+    EXPECT_EQ(input_A->getShape(), shape_t({1, 10}));
+    auto& output = outputs.at(pipelineOutputName);
+    EXPECT_EQ(output->getShape(), shape_t({0, 1, 10}));
+
+    std::shared_ptr<ModelInstance> modelInstance;
```

Comment:
Why do we need to check dummy shape?

Reply:
If to detect possibly edits in metadata after metadata call, please insert the comment to explain why

Reply:
This only checks model metadata after predict - metadata call can also affect model inputs/outputs

Reply:
The bug was that metadata call changed the metadata itself. The test verifies this call 2 times and original metadata of model that were changed before